src/hotspot/share/gc/g1/g1BlockOffsetTable.cpp

```diff
@@ -211,55 +211,34 @@ void G1BlockOffsetTable::update_for_block_work(HeapWord* blk_start, HeapWord* bl
          "boundary: " PTR_FORMAT,
          (uint)offset_array(offset_card),
          p2i(blk_start), p2i(boundary));
-  for (uint8_t* j = offset_card + 1; j <= end_card; j++) {
-    assert(offset_array(j) > 0 &&
-           offset_array(j) <= (uint8_t) (CardTable::card_size_in_words() + BOTConstants::N_powers - 1),
-           "offset array should have been set - "
-           "%u not > 0 OR %u not <= %u",
-           (uint) offset_array(j),
-           (uint) offset_array(j),
-           (uint) (CardTable::card_size_in_words() + BOTConstants::N_powers - 1));
-  }
 #endif
+
+  debug_only(G1BlockOffsetTable::verify_for_block(blk_start, blk_end);)
```

Comment:
I would prefer to make the `verify_for_block` method have a `NOT_DEBUG_RETURN` and enclose the actual method with an `#ifdef ASSERT` - this avoids the `debug_only` macro use which is kind of distracting imo.

Reply:
Fixed. There is a `ASSERT` block above `G1BlockOffsetTable::verify_for_block(blk_start, blk_end);` so I put it into the `ASSERT` block.